main.py

```diff
@@ -1,65 +1,340 @@
+
+        index = {}
+        for row in reader:
+            if _type == 'plates_file':
+                index[row['state'] + row['plate']] = row
+            elif _type == 'license_file':
+                index[row['ticket_number']] = row['anon_ticket_number']
+            elif _type == 'centriod_file':
+                index[row['SEG_ID']] = row
+            else:
+                raise Exception('`{}` not a supported index file type'.format(_type))
+
+        return index
+
+def save_index_file(path, index, headers, _type):
+    if path:
+        with smart_open(path, 'w') as file:
+            writer = csv.DictWriter(file, fieldnames=headers)
+
+            writer.writeheader()
+
+            for key, value in index.items():
+                if _type == 'plates_file':
+                    writer.writerow(value)
+                elif _type == 'license_file':
+                    writer.writerow({'ticket_number': key, 'anon_ticket_number': value})
+                else:
+                    raise Exception('`{}` not a supported index file type'.format(_type))
+
+@click.command()
+@click.option('--plates-file')
+@click.option('--ticket-numbers-file')
+@click.option('--centriod-file')
+@click.option('--latlon-input/--no-latlon-input', is_flag=True, default=True)
+@click.option('--deduplicate/--no-deduplicate', is_flag=True, default=False, help="Add 'E' to the end of duplicate ticket numbers")
+def main(plates_file, ticket_numbers_file, centriod_file, latlon_input, deduplicate):
+    global logger, plates, plates_counter, ticket_numbers, ticket_numbers_counter, centriods, dedup
+
+    logger = get_logging()
+
+    dedup = deduplicate
+
+    plates = load_index_file(plates_file, 'plates_file')
+    plate_numbers_values = plates.values()
+    if len(plate_numbers_values) > 0:
+        plates_counter = max(map(lambda x: int(x['id']), plate_numbers_values))
     else:
-        return issuing_agency
-
-# Main
-table = etl.fromtext(strip=False)\
-            .rowmap(unpack_line, header=header, failonerror=True)\
-            .convert('fine', float)\
-            .addfield('issue_date_and_time', create_date_time)\
-            .select('{issue_date_and_time} >= "2012-01-01" and {fine} > 0')\
-            .convert('division', remove_ppa_division, pass_row=True)\
-            .convert('issuing_agency', fix_prisons_agency)
-print(etl.dicts(table))
+        plates_counter = 0
+
+    logger.info('Plates autoincrement starting at: {}'.format(plates_counter))
+
+    ticket_numbers = load_index_file(ticket_numbers_file, 'license_file')
+    ticket_numbers_values = ticket_numbers.values()
+    if len(ticket_numbers_values) > 0:
+        ticket_numbers_counter = max(map(lambda x: int(x), ticket_numbers_values))
+    else:
+        ticket_numbers_counter = 0
+
+    logger.info('Ticket number autoincrement starting at: {}'.format(ticket_numbers_counter))
+
+    centriods = load_index_file(centriod_file, 'centriod_file')
+
+    (
+        petl
```

Comment:
What is this input to this statement? I think by default petl pulls from stdin, but since the input files are being passed as arguments, I assume there is no stdin. Oh, unless the stdin is the new tickets file we receive? So we're giving input as both stdin and as args? If so, is there a way to at least add a helper in the `@click` decorators above so it shows in help text?

Reply:
I think an example in the README will clear this up